pkg/crc/oc/oc.go

```diff
@@ -39,10 +41,10 @@ func (oc Config) runCommand(isPrivate bool, args ...string) (string, string, err
 	}
 
 	if isPrivate {
-		return oc.Runner.RunPrivate(oc.OcExecutablePath, args...)
+		return oc.Runner.RunPrivate("timeout", append([]string{timeout, oc.OcExecutablePath}, args...)...)
```

Comment:
Instead of doing this better to use the oc command timeout option.
```
$ oc options
[...]
--request-timeout='0': The length of time to wait before giving up on a single server request. Non-zero values
should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests.
```

Reply:
Yes I tried but it's not global timeout and even if I put 5s, sometimes oc takes 20s.

Reply:
which version of oc (checked with oc-4.6.x and 4.7)  you tried it with? It does works for me

```
$ oc version --client
Client Version: 4.7.0-rc.2 

$ time oc get nodes
Unable to connect to the server: dial tcp 192.168.130.11:6443: connect: no route to host

real	0m3.198s
user	0m0.166s
sys	0m0.044s

$ time oc get nodes --request-timeout='1s'
Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

real	0m1.105s
user	0m0.130s
sys	0m0.038s
```

Reply:
I am using oc 4.6.16. 

```
INFO ssh 12.971452547s sudo systemctl start kubelet 
INFO ssh 229.354148ms oc get nodes --request-timeout 5s --context admin --cluster crc --kubeconfig /opt/kubeconfig 
INFO ssh 226.358692ms oc get nodes --request-timeout 5s --context admin --cluster crc --kubeconfig /opt/kubeconfig 
INFO ssh 20.551353534s oc get nodes --request-timeout 5s --context admin --cluster crc --kubeconfig /opt/kubeconfig 
INFO ssh 2.580351017s oc get nodes --request-timeout 5s --context admin --cluster crc --kubeconfig /opt/kubeconfig 
INFO ssh 264.985137ms oc get secret --request-timeout 5s --context admin --cluster crc --kubeconfig /opt/kubeconfig 
```

Reply:
Alright,I also tried it and looks like when it trying the initial request, does fail for timeouts.